lib/rspec/expectations/fail_with.rb

```diff
@@ -24,8 +24,7 @@ def fail_with(message, expected=nil, actual=nil)
                                "appropriate failure_message[_when_negated] method to return a string?"
         end
 
-        diff = differ.diff(actual, expected)
-        message = "#{message}\nDiff:#{diff}" unless diff.empty?
+        message = ::RSpec::Matchers::ExpectedsForMultipleDiffs.from(expected).message_with_diff(message, method(:differ), actual)
```

Comment:
I think, we can leave this `differ_factory` untouched, and a bit later I will just take a stab at refactoring `Differ` class.